airflow/providers/openlineage/extractors/manager.py

```diff
@@ -113,6 +115,10 @@ def extract_metadata(self, dagrun, task, complete: bool = False, task_instance=N
                 self.log.warning(
                     "Failed to extract metadata using found extractor %s - %s %s", extractor, e, task_info
                 )
+        elif IS_AIRFLOW_2_10_OR_HIGHER and (hook_lineage := self.get_hook_lineage()) is not None:
+            inputs, outputs = hook_lineage
+            task_metadata = OperatorLineage(inputs=inputs, outputs=outputs)
+            return task_metadata
```

Comment:
Shouldn't we also try to use hook-level lineage's inputs and outputs when there is an extractor but inputs/outputs are missing, before we move to inlets and outlets there?